### DISCUSSION TITLE
DX issue: Partial card generated didn't load

### DISCUSSION THREAD

**Discussion by cmaxw** [avo-hq/avo#3185] [Feature feedback]
I used the rails generator to generate a partial card for one of my resources. When I added it to the resource, it wouldn't load.

I clicked the link that says "This is not an issue with Avo. Use [this page](http://localhost:3000/portal/resources/Avo::Resources::Meetup/cards/book-club-2024.video_embed?index=0) to see why this frame failed to load."

It showed a long backtrace that went into the bowels of Avo's Card rendering. Better Errors clearly wasn't helping here.

Then, I looked at my logs. It showed that it had generated a view with `@dashboard.id` called. Since I put this card into a resource, `@dashboard` was `nil` and it caused the error.

So, I recommend you either conditionally call `@dashboard.id` or `@resource.id` to avoid the nil issue, or take it out altogether and put the instructions into the generated view as text.

**Comment by adrianthedev** [comment]
Agreed! That's a bug in the template. Moving this to an issue so we can add it to the queue.
Thanks, Charles!
